Sources/SwiftGen/Logs.swift

```diff
@@ -39,17 +39,38 @@ enum ANSIColor: UInt8, CustomStringConvertible {
   }
 }
 
+// Based on https://github.com/realm/SwiftLint/blob/0.39.2/Source/SwiftLintFramework/Extensions/QueuedPrint.swift
+
 func logMessage(_ level: LogLevel, _ string: CustomStringConvertible) {
-  switch level {
-  case .info:
-    fputs(ANSIColor.green.format("\(string)\n"), stdout)
-  case .warning:
-    fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stderr)
-  case .error:
-    fputs(ANSIColor.red.format("swiftgen: error: \(string)\n"), stderr)
+  outputQueue.async {
+    switch level {
+    case .info:
+      fputs(ANSIColor.green.format("\(string)\n"), stdout)
+    case .warning:
+      fputs(ANSIColor.yellow.format("swiftgen: warning: \(string)\n"), stderr)
+    case .error:
+      fputs(ANSIColor.red.format("swiftgen: error: \(string)\n"), stderr)
+    }
   }
 }
 
+private let outputQueue: DispatchQueue = {
```

Comment:
nit: I'd call it `logQueue` instead (and adapt the label accordingly too), as "output" in the context of SwiftGen can be misleading and make me thing about the code we generate, not what we print to stdout